### PR TITLE
Sorting and Pagination for all rest controllers

### PR DESCRIPTION
**Pull request by zhenghung**
Handle cleanup for world filter in the mongoOperationsService class, allowing me to pass in undefined queries.

### CODE REVIEW

File: src/modules/rest/controllers/common/rest.common.queries.ts [hunk 1]
```diff
@@ -1,27 +1,7 @@
...
-        required: false,
-        type: Number,
-    },
-    {
-        name: 'pageSize',
-        required: false,
-        type: Number,
-    },
+    WORLD_IMPLICIT_QUERY,
+    ...PAGINATION_IMPLICIT_QUERIES,
```
Comment (Maelstromeous) [inline src/modules/rest/controllers/common/rest.common.queries.ts:6]: Ha didn't think of that, nice one 👍

File: src/modules/rest/controllers/rest.instance.facility.control.controller.ts [hunk 2]
```diff
@@ -18,15 +20,21 @@ export default class RestInstanceFacilityControlController {
...
+            @Query('order') order?: string,
+            @Query('page', OptionalIntPipe) page?: number,
+            @Query('pageSize', OptionalIntPipe) pageSize?: number,
+    ): Promise<InstanceFacilityControlEntity[]> {
+        return await this.mongoOperationsService.findMany(InstanceFacilityControlEntity, {instance}, new Pagination({sortBy, order, page, pageSize}));
     }
 
     @Get(':instance/facility/:facility')
-    @ApiOperation({summary: 'Return a filtered list of InstanceFacilityControlEntity for an instance'})
+    @ApiOperation({summary: 'Return a single InstanceFacilityControlEntity of an instance'})
```
Comment (Maelstromeous) [inline src/modules/rest/controllers/rest.instance.facility.control.controller.ts:34]: 👍

File: src/modules/rest/controllers/rest.instance.metagame.controller.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import {ApiOperation, ApiResponse, ApiTags} from '@nestjs/swagger';
 import InstanceMetagameTerritoryEntity from '../../data/entities/instance/instance.metagame.territory.entity';
 import MongoOperationsService from '../../../services/mongo/mongo.operations.service';
 import {Ps2alertsEventState} from '../../data/constants/eventstate.consts';
-import {OptionalIntPipe} from '../pipes/OptionalIntPIpe';
+import {OptionalIntPipe} from '../pipes/OptionalIntPipe';
```
Comment (Maelstromeous) [inline src/modules/rest/controllers/rest.instance.metagame.controller.ts:6]: Strange how this didn't error 🤔

File: src/modules/rest/controllers/rest.instance.metagame.controller.ts [hunk 4]
```diff
@@ -73,12 +68,6 @@ export class RestInstanceMetagameController {
...
             @Query('pageSize', OptionalIntPipe) pageSize?: number,
     ): Promise<InstanceMetagameTerritoryEntity[]> {
-        const worldObject: {[k: string]: any} = {};
-
-        if (world) {
-            worldObject.world = world;
-        }
-
-        return await this.mongoOperationsService.findMany(InstanceMetagameTerritoryEntity, worldObject, new Pagination({sortBy, order, page, pageSize}));
+        return await this.mongoOperationsService.findMany(InstanceMetagameTerritoryEntity, {world}, new Pagination({sortBy, order, page, pageSize}));
```
Comment (Maelstromeous) [inline src/modules/rest/controllers/rest.instance.metagame.controller.ts:71]: Presumably this has been applied elsewhere?

**Review by Maelstromeous — APPROVED** [review]
Noice

File: src/modules/rest/controllers/rest.instance.metagame.controller.ts (hunk 3 above)
Comment (zhenghung) [reply src/modules/rest/controllers/rest.instance.metagame.controller.ts:6]: Typo was present in the filename.

File: src/modules/rest/controllers/rest.instance.metagame.controller.ts (hunk 4 above)
Comment (zhenghung) [reply src/modules/rest/controllers/rest.instance.metagame.controller.ts:71]: Yea the mongoOperationService checks for empty objects and deletes keys with undefined values. So we don't need to handle it here

Comment (Maelstromeous) [reply src/modules/rest/controllers/rest.instance.metagame.controller.ts:71]: Nice, good refactor

File: src/modules/rest/controllers/rest.instance.metagame.controller.ts (hunk 3 above)
Comment (Maelstromeous) [reply src/modules/rest/controllers/rest.instance.metagame.controller.ts:6]: Aha that would be it!

**Pull request merged**